stdlib/2and3/_typeshed/__init__.pyi

```diff
@@ -15,12 +15,28 @@
 import array
 import mmap
 import sys
-from typing import Protocol, Text, TypeVar, Union
+from typing import AbstractSet, Container, Protocol, Text, Tuple, TypeVar, Union
 from typing_extensions import Literal
 
+_KT_co = TypeVar("_KT_co", covariant=True)
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT = TypeVar("_VT")
+_VT_co = TypeVar("_VT_co", covariant=True)
 _T_co = TypeVar("_T_co", covariant=True)
 _T_contra = TypeVar("_T_contra", contravariant=True)
 
+# Mapping-like protocols
+
+class SupportsItems(Protocol[_KT_co, _VT_co]):
+    def items(self) -> AbstractSet[Tuple[_KT_co, _VT_co]]: ...
+
+class SupportsGetItem(Container[_KT_contra], Protocol[_KT_contra, _VT_co]):
+    def __getitem__(self, __k: _KT_contra) -> _VT_co: ...
```

Comment:
See python/typing#273 and #550 for some previous discussion on the variance of Mapping (not directly relevant here).